backend/src/stwo/circuit_builder.rs

```diff
@@ -44,7 +44,7 @@ where
                 coset_index_to_circle_domain_index(i, slice.len().ilog2()),
                 slice.len().ilog2(),
             ),
-            v.try_into_i32().unwrap().into(),
+            v.to_integer().try_into_u32().unwrap().into(),
```

Comment:
This is how we convert our field elements to theirs, right? That's why it doesn't matter if we go via u32 or i32?

Reply:
Exactly, going through i32 was wrong